vignettes/tcplfit2-vignette.Rmd

```diff
@@ -161,3 +166,159 @@ The input data for this example contains 6 signatures for one chemical in a tran
+dat[,cval := rval]
+
+## Poor well quality (wllq) wells should be removed
+dat <- dat[!wllq == 0,]
+
+## Fitting generally cannot occur if response values are NA therefore values need to be removed
+dat <- dat[!is.na(cval),]
+
+## A column for log10 concentration is added as some of the mc3 methods require logc. Given logging concentration, conc=0 are not allowed therefore a dummy non-zero value should be used
+dat[conc == 0 , conc := 0.01]
+dat[ , logc := log10(conc)]
+
+#As a final step to prepare the dataset tcplfit2 processing, a dummy aeid is required if using mc3_mthds from tcpl
+dummy_aeid <- 99999
+dat[,aeid := dummy_aeid]
+
+## Set aeid as a key
+setkey(dat,aeid)
+```
+
+Once the data is initialized to a point where the required fields are available, the methods included in the `tcpl` package can be identifued and applied without the need for a database connection. You can see the list of available methods for Level 3 in the table below:
+
+```{r example4_mthdlist, fig.height = 6, fig.width = 7, warning = FALSE, echo = FALSE}
+mthd_funcs <- tcpl:::mc3_mthds()
+DT::datatable(tcpl::tcplMthdList(3),rownames= FALSE, options = list(scrollX = T))
+```
+
+Here three normalization methods are selected and applied to the data.  Note because of the way `tcpl` handles the application of functions, the dataframe must be called `dat`. In the future, `tcpl` will export these functions so that they can be applied to any dataset without the need for a specific name or dummy aeid.
+
+```{r example4_mc3methods, fig.height = 6, fig.width = 7, results = 'hide'}
+# apply level 3 methods
+lapply(mthd_funcs[["bval.apid.nwlls.med"]](dummy_aeid), eval)
+lapply(mthd_funcs[["pval.apid.medncbyconc.min"]](dummy_aeid),eval)
+lapply(mthd_funcs[["resp.pc"]](dummy_aeid),eval)
+```
+
+Level 4 determines the baseline variability, or noise, that will later be used for cutoff calculation. Using the established concentration index, the level 4 methods can be loaded in a similar way to level 3.
+
+```{r example4_mthdlist_l4, fig.height = 6, fig.width = 7}
+mthd_funcs_l4 <- tcpl:::mc4_mthds()
+DT::datatable(tcpl::tcplMthdList(4), rownames= FALSE, options = list(scrollX = T))
+```
+
+There are much fewer level 4 methods, but generally it is a requirement to assign a method that calculates the bmad and assign a method that calculates the standard deviation of the noise for `tcplfit2` fitting.
+
+```{r example4_mc4methods, fig.height = 6, fig.width = 7, results = 'hide'}
+# apply level 4 methods
+lapply(mthd_funcs_l4[["bmad.aeid.lowconc.twells"]](),eval)
+lapply(mthd_funcs_l4[["onesd.aeid.lowconc.twells"]](),eval)
+lapply(mthd_funcs_l4[["bidirectional.false"]](),eval)
+```
+
+After methods up to level 4 have been applied, the model fitting can begin. In `tcpl`, this would be considered level 4, and is where `tcplfit2` is used to fit all of the models as a dependency of `tcpl`.
```

Comment:
Should this be "dependency for tcpl"?  If not, then ignore this comment.

Reply:
I'm not sure what way is correct so if you think dependency for tcpl sounds better I will switch it to that.